src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,96 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Todo = {
+  userId: number;
+  id: number;
+  title: string;
+  completed: boolean;
+};
+
+type Props = {
+  modalShowId: number;
+  setModalShowId: (value: number) => void;
+  todos: Todo[];
+};
+
+type User = {
+  id: number;
+  name: string;
+  email: string;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  modalShowId,
+  setModalShowId,
+  todos,
+}) => {
+  const [loading, setLoading] = useState(true); // Состояние для индикации загрузки
+  const [user, setUser] = useState<User | null>(null); // Состояние для данных пользователя
+
+  const todo = todos.find(cTodo => cTodo.id === modalShowId); // Поиск задачи по ID
+
+  // Загружаем данные пользователя при наличии todo
```

Comment:
Remove comment. Try to write obvious code that explains all we need without comments)